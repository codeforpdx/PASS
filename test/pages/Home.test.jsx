import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { expect, describe, it, afterEach } from 'vitest';
import { Home } from '@pages';
import createMatchMedia from '../helpers/createMatchMedia';

describe('Home Page', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders with correct order of logoSection on mobile', () => {
    window.matchMedia = createMatchMedia(599);
    render(<Home />);
    const h1 = screen.getByTestId('testStack');
    const children = Array.from(h1.children);

    expect(children.length).toBe(3);
    expect(children[0].tagName.toLowerCase()).toBe('span');
    expect(children[1].tagName.toLowerCase()).toBe('img');
    expect(children[2].tagName.toLowerCase()).toBe('span');
  });

  it('rrenders with correct order of logoSection on desktop', () => {
    window.matchMedia = createMatchMedia(1200);
    render(<Home />);
    const h1 = screen.getByTestId('testStack');
    const children = Array.from(h1.children);

    expect(children.length).toBe(2);
    expect(children[0].tagName.toLowerCase()).toBe('span');
    expect(children[1].tagName.toLowerCase()).toBe('span');
  });
});
