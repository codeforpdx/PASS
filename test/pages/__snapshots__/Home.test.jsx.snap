// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Page > renders 1`] = `
<div>
  <main
    class="MuiContainer-root MuiContainer-maxWidthLg css-1sj9by6-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-1gqbj6n"
    >
      <section
        id="home"
      >
        <div
          class="MuiStack-root css-1z0rh7t-MuiStack-root"
        >
          <img
            alt=""
            class="MuiBox-root css-1v5z18m"
            src="/assets/web-security-green.png"
          />
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1loew7y-MuiTypography-root"
          >
            <strong>
              Keep Your Documents Safe and Secure Using Decentralized Technology
            </strong>
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1 css-acfnqh-MuiTypography-root"
          >
            Our innovative solution empowers individuals to manage their critical documents and control access for trusted organizations. PASS simplifies service access, enabling seamless documents requests and secure data sharing for a smoother support process.
          </p>
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1fdjaxb-MuiButtonBase-root-MuiButton-root"
            href="mailto:hugh@codeforpdx.org"
            tabindex="0"
          >
            Request a Demo
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <img
            alt=""
            class="MuiBox-root css-1v5z18m"
            src="/assets/app-green.png"
          />
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1loew7y-MuiTypography-root"
          >
            <strong>
              An App for Caseworkers
            </strong>
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1 css-acfnqh-MuiTypography-root"
          >
            PASS allows users to quickly and securely share documents of their clients within the app. The app helps caseworkers verify and share documents such as ID and proof of income while retaining total control of them.
          </p>
          <img
            alt=""
            class="MuiBox-root css-1v5z18m"
            src="/assets/key-features-green.png"
          />
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1loew7y-MuiTypography-root"
          >
            <strong>
              Key Features
            </strong>
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1 css-acfnqh-MuiTypography-root"
          />
          <div
            class="MuiBox-root css-1tohi9w"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
              data-testid="SecurityIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 1 3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm0 10.99h7c-.53 4.12-3.28 7.79-7 8.94V12H5V6.3l7-3.11v8.8z"
              />
            </svg>
            <strong>
              Secure Storage
            </strong>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-lfz6e5-MuiTypography-root"
          >
            Store vital documents like IDs, Social Security information, birth certificates, medical records, and bank statements in a valid digital format.
          </p>
          <div
            class="MuiBox-root css-1tohi9w"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
              data-testid="Diversity1Icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="4"
                cy="14"
                r="2"
              />
              <path
                d="M1.22 17.58C.48 17.9 0 18.62 0 19.43V21h4.5v-1.61c0-.83.23-1.61.63-2.29-.37-.06-.74-.1-1.13-.1-.99 0-1.93.21-2.78.58z"
              />
              <circle
                cx="20"
                cy="14"
                r="2"
              />
              <path
                d="M22.78 17.58c-.85-.37-1.79-.58-2.78-.58-.39 0-.76.04-1.13.1.4.68.63 1.46.63 2.29V21H24v-1.57c0-.81-.48-1.53-1.22-1.85zm-6.54-.93c-1.17-.52-2.61-.9-4.24-.9-1.63 0-3.07.39-4.24.9C6.68 17.13 6 18.21 6 19.39V21h12v-1.61c0-1.18-.68-2.26-1.76-2.74zM9 12c0 1.66 1.34 3 3 3s3-1.34 3-3-1.34-3-3-3-3 1.34-3 3z"
              />
              <path
                d="M2.48 10.86C2.17 10.1 2 9.36 2 8.6 2 6.02 4.02 4 6.6 4c2.68 0 3.82 1.74 5.4 3.59C13.57 5.76 14.7 4 17.4 4 19.98 4 22 6.02 22 8.6c0 .76-.17 1.5-.48 2.26.65.31 1.18.82 1.53 1.44.6-1.2.95-2.42.95-3.7C24 4.9 21.1 2 17.4 2c-2.09 0-4.09.97-5.4 2.51C10.69 2.97 8.69 2 6.6 2 2.9 2 0 4.9 0 8.6c0 1.28.35 2.5.96 3.7.35-.62.88-1.13 1.52-1.44z"
              />
            </svg>
            <strong>
              Nonprofit-Caseworker Integration
            </strong>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-lfz6e5-MuiTypography-root"
          >
            The platform facilitates smooth communication between nonprofit organizations, case workers, and the individuals they serve. It allows nonprofit organizations to maintain a contact list, and caseworkers are assigned contacts whose data they can access securely.
          </p>
          <div
            class="MuiBox-root css-1tohi9w"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
              data-testid="SupportIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm7.46 7.12-2.78 1.15c-.51-1.36-1.58-2.44-2.95-2.94l1.15-2.78c2.1.8 3.77 2.47 4.58 4.57zM12 15c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zM9.13 4.54l1.17 2.78c-1.38.5-2.47 1.59-2.98 2.97L4.54 9.13c.81-2.11 2.48-3.78 4.59-4.59zM4.54 14.87l2.78-1.15c.51 1.38 1.59 2.46 2.97 2.96l-1.17 2.78c-2.1-.81-3.77-2.48-4.58-4.59zm10.34 4.59-1.15-2.78c1.37-.51 2.45-1.59 2.95-2.97l2.78 1.17c-.81 2.1-2.48 3.77-4.58 4.58z"
              />
            </svg>
            <strong>
              Support Service
            </strong>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-lfz6e5-MuiTypography-root"
          >
            Verified documents can be used to facilitate access to service such as housing support and shelter accommodation. The platform simplifies the process of submitting necessary documents for such services.
          </p>
        </div>
      </section>
    </div>
  </main>
</div>
`;
